pytorch_lightning/utilities/cli.py

```diff
@@ -332,6 +347,15 @@ def add_arguments_to_parser(self, parser: LightningArgumentParser) -> None:
 
     def link_optimizers_and_lr_schedulers(self) -> None:
         """Creates argument links for optimizers and learning rate schedulers that specified a ``link_to``."""
+        if any(
+            True for v in sys.argv for optim_name in OPTIMIZER_REGISTRIES.keys() if f"--optimizer={optim_name}" in v
```

Comment:
This does not consider:

1. Passing values without `=`: `--optimizer Adam`
2. That the optimizer key can change: `--opt=Adam`

Reply:
Great catch ! Done !